packages/@ourworldindata/grapher/src/mapCharts/MapChart.tsx

```diff
@@ -375,10 +362,42 @@ export class MapChart
         return [bins[bins.length - 1], ...bins.slice(0, -1)]
     }
 
+    @computed private get numMembersPerCategoricalBinByIndex(): Map<
+        number,
+        number
+    > {
+        const { chartState, legendData } = this
+        return new Map(
+            legendData
+                .filter((bin) => isCategoricalBin(bin))
+                .map((bin) => {
+                    const memberCount = chartState.series.filter((series) =>
+                        bin.contains(series.value)
+                    ).length
+                    return [bin.index, memberCount]
+                })
+        )
+    }
+
     @computed get categoricalLegendData(): CategoricalBin[] {
-        return this.legendData
+        let categoricalLegendData = this.legendData
             .filter((bin) => isCategoricalBin(bin))
             .map((bin) => this.maybeAddPatternRefToBin(bin))
+
+        // Hide empty bins for static charts
+        if (this.isStatic) {
+            categoricalLegendData = categoricalLegendData.filter((bin) => {
+                const memberCount =
+                    this.numMembersPerCategoricalBinByIndex.get(bin.index) ?? 0
+                return (
+                    isNoDataBin(bin) ||
+                    isProjectedDataBin(bin) ||
+                    memberCount > 0
```

Comment:
So here:

http://staging-site-rich-data-revisions/grapher/covid-vaccination-policy.png?tab=map&imType=thumbnail&imWidth=1200&imHeight=640&nocache=true

I would have thought that "One vulnerable group" wouldn't show up here, but it does, because it's technically categorical data? 

And then I guess you don't want to hide numeric bins even if they're empty to avoid people getting confused about which numeric bin a colour represents?

Reply:
That’s right! It’s technically numerical data, and I don’t think we’d want to hide numerical bins since they’re usually sequential, if you know what I mean.

We could technically detect numerical-bins-that-look-like-categorical-ones, but that could get nasty. The main reason I added this is that categorical legend items otherwise take up too much space in thumbnails (e.g. http://staging-site-rich-data-revisions/grapher/leading-cancer-types-causing-death-in-women.svg?tab=map&imType=thumbnail&imFontSize=12). Numerical legends are always just a single line, so there isn’t that much pressure to optimise there.